src/engraving/infrastructure/mscwriter.cpp

```diff
@@ -266,7 +284,11 @@ Ret MscWriter::ZipFileWriter::open(io::IODevice* device, const path_t& filePath)
 {
     m_device = device;
     if (!m_device) {
-        m_device = new File(filePath);
+        // Create a corrupted file if the filename contains " - CORRUPTED.mscz".
+        // This is only for DEV/QA testing and will not be included in a release.

```

Comment:
We can enable it for dev/QA by using IGlobalConfiguration::devModeEnabled (which can be set even in release builds via DevTools/Settings). I also think that instead of creating FileCorruptor directly here, we can pass it as a device to open()